build.gradle

```diff
@@ -153,7 +158,24 @@ task renameShadowJar(type: Copy, group: 'release', dependsOn: [shadowJar]) {
 task updateAssets(dependsOn: ['gitCloneAssets', 'gitPullAssets']) {
 }
 
-task allPlatform(type: Zip, group: 'release', dependsOn: [renameShadowJar, updateAssets]) {
+task prepareGameEngineProperties(group: 'release') {
+    inputs.file gameEnginePropertiesFile
+    outputs.file gameEnginePropertiesArtifactFile
+
+    doLast {
+        copy {
+            from gameEnginePropertiesFile
+            into gameEnginePropertiesArtifactFile.parent
+        }
+        ant.propertyfile(file: gameEnginePropertiesArtifactFile) {
+            entry key: 'engine_version', value: version
+        }
+    }
+}
+
+task prepareRootFiles(group: 'release', dependsOn: [prepareGameEngineProperties]) {}
```

Comment:
I think I started working top-down and just assumed I would do everything root-file-related in `prepareRootFiles`.  Then when I made the task incremental, it didn't make sense anymore, and I broke out `prepareGameEngineProperties` thinking there could be more root file processing in the future.  Since there isn't any now, I went ahead and removed `prepareRootFiles`.  We can always add it back if a second root file comes along. :smile: bafbc8c